documentation/modules/ROOT/pages/operations/quarkus-debezium-engine-extension.adoc

```diff
@@ -0,0 +1,422 @@
+        // process your events
+    }
+
+}
+```
+
+The `CapturingEvent<T>` contains information related to the kind of database operation:
+
+```java
+    @Capturing
+    public void capture(CapturingEvent<SourceRecord> record) {
+        switch (record) {
+            case Create<SourceRecord> event -> {}
+            case Delete<SourceRecord> event -> {}
+            case Message<SourceRecord> event -> {}
+            case Read<SourceRecord> event -> {}
+            case Truncate<SourceRecord> event -> {}
+            case Update<SourceRecord> event -> {}
+        }
+    }
+```
+
+=== Filtering Change Data Capture Events
+
+It's possible to filter events by `destination`:
+
+```java
+    @Capturing(destination = "native.inventory.products")
+    public void capture(CapturingEvent<SourceRecord> record) {
+        // process your event
+    }
+```
+
+The default behavior is that a Debezium connector `destination` is formed from the name of the `prefix` defined in the configuration with the database name and the name of the table in which the change was made. In some cases the `destination` is redefined using an https://debezium.io/documentation/reference/stable/transformations/topic-routing.html[SMT].
+
+=== Deserializer via Jackson
+
+Quarkus has built-in support for JSON serialization and deserialization based on Jackson. There is an existing `ObjectMapperDeserializer` that can be used to deserialize all data objects via Jackson.
+
+The corresponding deserializer class needs to be subclassed. So, we have to create a `ProductDeserializer` that extends the `ObjectMapperDeserializer`.
+
+```java
+public class ProductDeserializer extends ObjectMapperDeserializer<Product> {
+    public ProductDeserializer() {
+        super(Product.class);
+    }
+}
+```
+
+Finally, configure your capture channel to use the Jackson deserializer for a particular destination:
+
+```properties
+quarkus.debezium.capturing.products.destination=native.inventory.products
+quarkus.debezium.capturing.products.deserializer=com.acme.product.jackson.ProductDeserializer
+```
+
+and use it in your code:
+
+```java
+import io.debezium.runtime.CapturingEvent;
+import jakarta.enterprise.context.ApplicationScoped;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import io.debezium.runtime.Capturing;
+
+@ApplicationScoped
+public class ProductHandler {
+
+
+    @Capturing(destination = "native.inventory.products")
+    public void capture(CapturingEvent<Product> record) {
+        // process your events
+    }
+
+}
+```
+
+or if you don't need information related to the database operation:
+
+```java
+import io.debezium.runtime.CapturingEvent;
+import jakarta.enterprise.context.ApplicationScoped;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import io.debezium.runtime.Capturing;
+
+@ApplicationScoped
+public class ProductHandler {
+
+
+    @Capturing(destination = "native.inventory.products")
+    public void capture(Product product) {
```

Comment:
I will add a note that in this case the event is flattened and so the information on the kind of operation is lost.

Reply:
Oh I missed it but I have the fear that as me also users can miss it. Maybe a `NOTE` section could evidence it more.